thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/ThirdEyeAnomalyApplication.java

```diff
@@ -73,45 +73,34 @@ public void run(final ThirdEyeAnomalyConfiguration config, final Environment env
     super.initDetectorRelatedDAO();
     ThirdEyeCacheRegistry.initializeDetectorCaches(config);
 
-    final AnomalyFunctionFactory anomalyFunctionFactory =
-        new AnomalyFunctionFactory(config.getFunctionConfigPath());
-
-    if (config.isScheduler()) {
-      detectionJobScheduler = new DetectionJobScheduler(anomalyJobDAO, anomalyTaskDAO,
-          anomalyFunctionDAO);
-    }
-    if (config.isWorker()) {
-      taskDriver =
-        new TaskDriver(config.getId(), anomalyJobDAO, anomalyTaskDAO, anomalyResultDAO,
-            anomalyFunctionRelationDAO, anomalyFunctionFactory, hibernateBundle.getSessionFactory());
-    }
-    if (config.isMonitor()) {
-      monitorJobScheduler = new MonitorJobScheduler(anomalyJobDAO, anomalyTaskDAO,
-          config.getMonitorConfiguration());
-    }
-
-
     environment.lifecycle().manage(new Managed() {
       @Override
       public void start() throws Exception {
-        new HibernateSessionWrapper<Void>(hibernateBundle.getSessionFactory())
-            .execute(new Callable<Void>() {
-          @Override
-          public Void call() throws Exception {
-            if (config.isWorker()) {
-              taskDriver.start();
-            }
-            if (config.isScheduler()) {
-              detectionJobScheduler.start();
-              environment.jersey()
-              .register(new DetectionJobResource(detectionJobScheduler, anomalyFunctionDAO));
-            }
-            if (config.isMonitor()) {
-              monitorJobScheduler.start();
-            }
-            return null;
-          }
-        });
+
+        if (config.isWorker()) {
+          anomalyFunctionFactory = new AnomalyFunctionFactory(config.getFunctionConfigPath());
+          taskDriver =
+              new TaskDriver(config, anomalyJobDAO, anomalyTaskDAO, anomalyResultDAO,
+                  anomalyFunctionRelationDAO, anomalyFunctionFactory, hibernateBundle.getSessionFactory());
```

Comment:
SessionFactory was being used by AnomalyFunctionRelationDAO. I removed the use of relationDAO completely as it's not being used. We can consider adding it back if it comes of use later.